random errors downloading images

We're seeing some occasional errors fetching images, not sure if they're related to #83 but it might be the case.

[Semaphore](https://semaphoreci.com/coreos/rkt/branches/pull-request-2714/builds/3):

```
	rkt_tests.go:115: Running command: /home/runner/rkt/builds/build-rkt-coreos/build-rkt-1.7.0+git/tmp/functional/rkt --dir=/tmp/datadir-322458840 --local-config=/tmp/localdir-586074711 --system-config=/tmp/systemdir-459955146 --user-config=/tmp/userdir-702517409 --store-only --insecure-options=image fetch docker://busybox:latest 
	rkt_tests.go:216: Error: ExpectRegex didn't find regex 'sha512-[0-9a-f]{32,64}'.
		Output: image: remote fetching from URL "docker://busybox:latest"

		Downloading sha256:a3ed95caeb0 [===================================] 32 B / 32 B

		fetch: error generating ACI: error writing ACI: unexpected EOF
```

On fedora-rawhide:

```
17:48:13 	rkt_tests.go:115: Running command: /home/fedora/workspace/rkt-matrix/dummy/dummy1/label/fedora-rawhide-flavor-host/builds/build-rkt-host/build-rkt-1.6.0+git/tmp/functional/rkt --dir=/tmp/datadir-552332049 --local-config=/tmp/localdir-616853308 --system-config=/tmp/systemdir-999719275 --user-config=/tmp/userdir-726059470 --store-only --insecure-options=image fetch docker://busybox
17:48:13     rkt_tests.go:216: Error: ExpectRegex didn't find regex 'sha512-[0-9a-f]{32,64}'.
17:48:13         Output: image: remote fetching from URL "docker://busybox"
17:48:13         Downloading sha256:a3ed95caeb0 [===================================] 32 B / 32 B
17:48:13         fetch: error generating ACI: error writing ACI: flate: corrupt input before offset 45274
```

--- a/lib/internal/backend/repository/repository2.go
+++ b/lib/internal/backend/repository/repository2.go
@@ -26,6 +26,7 @@ import (
 	"regexp"
 	"strconv"
 	"strings"
+	"sync"
 	"time"
 
 	"github.com/appc/docker2aci/lib/common"
@@ -80,13 +81,17 @@ func (rb *RepositoryBackend) buildACIV2(layerIDs []string, dockerURL *types.Pars
 
 	var errChannels []chan error
 	closers := make([]io.ReadCloser, len(layerIDs))
+	var wg sync.WaitGroup
 	for i, layerID := range layerIDs {
-		errChan := make(chan error)
+		wg.Add(1)
+		errChan := make(chan error, 1)
 		errChannels = append(errChannels, errChan)
 		// https://github.com/golang/go/wiki/CommonMistakes
 		i := i // golang--
 		layerID := layerID
 		go func() {
+			defer wg.Done()
+
 			manifest := rb.imageManifests[*dockerURL]
 
 			layerIndex, err := getLayerIndex(layerID, manifest)
@@ -120,6 +125,8 @@ func (rb *RepositoryBackend) buildACIV2(layerIDs []string, dockerURL *types.Pars
 			errChan <- nil
 		}()
 	}
+	// Need to wait for all of the readers to be added to the copier (which happens during rb.getLayerV2)
+	wg.Wait()
 	err = copier.PrintAndWait(os.Stderr, 500*time.Millisecond, nil)
 	if err != nil {
 		return nil, nil, err

